edit/edit.go

```diff
@@ -461,6 +476,150 @@ func ContainsComments(expr build.Expr, str string) bool {
 	return false
 }
 
+// RemoveEmptySelectsAndConcatLists iterates the tree in order to turn
+// empty selects into empty lists and adjacent lists are concatenated
+func RemoveEmptySelectsAndConcatLists(e build.Expr) build.Expr {
+	switch e := e.(type) {
+	case *build.BinaryExpr:
+		if e.Op == "+" {
+			e.X = RemoveEmptySelectsAndConcatLists(e.X)
+			e.Y = RemoveEmptySelectsAndConcatLists(e.Y)
+
+			x, xIsList := e.X.(*build.ListExpr)
+			y, yIsList := e.Y.(*build.ListExpr)
+
+			if xIsList && yIsList {
+				return &build.ListExpr{List: append(x.List, y.List...)}
+			}
+
+			if xIsList && len(x.List) == 0 {
+				return e.Y
+			}
+
+			if yIsList && len(y.List) == 0 {
+				return e.X
+			}
+		}
+	case *build.CallExpr:
+		if x, ok := e.X.(*build.Ident); ok && x.Name == "select" {
+			dict := e.List[0].(*build.DictExpr)
```

Comment:
This will crash if the `select` call has no arguments or the first argument is not a dict expression. That shoudn't usually happen, but it's still not a syntax error and buildozer shouldn't crash in this case.

Also, the argument to the `select` function can be a constant defined elswhere: `select(MY_CONFIGS)`. Please make sure buildozer doesn't crash there (and add a test).